File: ischia/src/main/java/cz/fb/manaus/ischia/validator/update/TheSamePriceUpdateEpsilonValidator.java
package cz.fb.manaus.ischia.validator.update;

import cz.fb.manaus.ischia.BackLoserBet;
import cz.fb.manaus.ischia.LayLoserBet;
import cz.fb.manaus.reactor.betting.validator.common.update.AbstractTooCloseUpdateEpsilonValidator;
import org.springframework.context.annotation.Profile;
import org.springframework.core.Ordered;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

@BackLoserBet
@LayLoserBet
@Component
@Order(Ordered.HIGHEST_PRECEDENCE)
@Profile("matchbook")
public class TheSamePriceUpdateEpsilonValidator extends AbstractTooCloseUpdateEpsilonValidator {

    public TheSamePriceUpdateEpsilonValidator() {
        super(0.025);
    }

    @Override
    public boolean isDowngradeAccepting() {
        return false;
    }
}

File: ischia/src/main/java/cz/fb/manaus/ischia/validator/update/TheSamePriceUpdateEpsilonValidator.kt
package cz.fb.manaus.ischia.validator.update

import cz.fb.manaus.ischia.BackUniverse
import cz.fb.manaus.ischia.LayUniverse
import cz.fb.manaus.reactor.betting.validator.Validator
import cz.fb.manaus.reactor.betting.validator.common.update.TooCloseUpdateEpsilonValidator
import org.springframework.core.Ordered
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component

@BackUniverse
@LayUniverse
@Component
@Order(Ordered.HIGHEST_PRECEDENCE)
object TheSamePriceUpdateEpsilonValidator : Validator by TooCloseUpdateEpsilonValidator(0.025) {

    override val isDowngradeAccepting: Boolean = false

}
